service/matching/matching_engine.go

```diff
@@ -1372,34 +1378,99 @@ func (e *matchingEngineImpl) unloadTaskQueuePartition(unloadPM taskQueuePartitio
+		partitionType: pm.partitionMgr.Partition().Kind(),
+		versioned:     versioned,
+	}
+
+	e.gaugeMetricCounter.gaugeMetricCounterLock.Lock()
+	e.gaugeMetricCounter.loadedPhysicalTaskQueueCount[PhysicalTaskQueueParameters] += delta
+	loadedPhysicalTaskQueueCounter := e.gaugeMetricCounter.loadedPhysicalTaskQueueCount[PhysicalTaskQueueParameters]
+	e.gaugeMetricCounter.gaugeMetricCounterLock.Unlock()
+
+	pmImpl := pm.partitionMgr
+
+	e.metricsHandler.Gauge(metrics.LoadedPhysicalTaskQueueCounter.Name()).Record(
+		float64(loadedPhysicalTaskQueueCounter),
+		metrics.NamespaceTag(pmImpl.ns.Name().String()),
+		metrics.TaskTypeTag(PhysicalTaskQueueParameters.taskType.String()),
+		metrics.PartitionTypeTag(PhysicalTaskQueueParameters.partitionType.String()),
+	)
+}
+
+// Responsible for emitting and updating loaded_task_queue_family_count, loaded_task_queue_count and
+// loaded_task_queue_partition_count metrics
+func (e *matchingEngineImpl) updateTaskQueuePartitionGauge(pm taskQueuePartitionManager, delta int) {
+
+	// each metric shall be accessed based on the mentioned parameters
+	TaskQueueFamilyParameters := taskQueueCounterKey{
+		namespaceID: pm.Partition().NamespaceId(),
+	}
+
+	TaskQueueParameters := taskQueueCounterKey{
 		namespaceID: pm.Partition().NamespaceId(),
 		taskType:    pm.Partition().TaskType(),
-		kind:        pm.Partition().Kind(),
-		versioned:   versioned,
 	}
 
-	e.taskQueueCountLock.Lock()
-	e.taskQueueCount[countKey] += delta
-	newCount := e.taskQueueCount[countKey]
-	e.taskQueueCountLock.Unlock()
+	TaskQueuePartitionParameters := taskQueueCounterKey{
+		namespaceID:   pm.Partition().NamespaceId(),
+		taskType:      pm.Partition().TaskType(),
+		partitionType: pm.Partition().Kind(),
+	}
+
+	rootPartition := pm.Partition().IsRoot()
+	e.gaugeMetricCounter.gaugeMetricCounterLock.Lock()
+
+	loadedTaskQueueFamilyCounter, loadedTaskQueueCounter, loadedTaskQueuePartitionCounter :=
+		e.gaugeMetricCounter.loadedTaskQueueFamilyCount[TaskQueueFamilyParameters], e.gaugeMetricCounter.loadedTaskQueueCount[TaskQueueParameters],
+		e.gaugeMetricCounter.loadedTaskQueuePartitionCount[TaskQueuePartitionParameters]
 
-	nsEntry, err := e.namespaceRegistry.GetNamespaceByID(countKey.namespaceID)
-	ns := namespace.Name("unknown")
-	if err == nil {
-		ns = nsEntry.Name()
+	loadedTaskQueuePartitionCounter += delta
+	e.gaugeMetricCounter.loadedTaskQueuePartitionCount[TaskQueuePartitionParameters] = loadedTaskQueuePartitionCounter
+	if rootPartition {
+		loadedTaskQueueCounter += delta
+		e.gaugeMetricCounter.loadedTaskQueueCount[TaskQueueParameters] = loadedTaskQueueCounter
+		if pm.Partition().TaskType() == enumspb.TASK_QUEUE_TYPE_WORKFLOW {
+			loadedTaskQueueFamilyCounter += delta
+			e.gaugeMetricCounter.loadedTaskQueueFamilyCount[TaskQueueFamilyParameters] = loadedTaskQueueFamilyCounter
+		}
 	}
+	e.gaugeMetricCounter.gaugeMetricCounterLock.Unlock()
+
+	pmImpl := pm.(*taskQueuePartitionManagerImpl)
+
+	// emitting all the three metrics, for now.
```

Comment:
why "for now"? what's going to change in the future? should it be commented with a TODO?